### ISSUE TITLE
VS2017 does not work with CMake files, EventPipe directory config is broken

### ISSUE THREAD

**Issue by daniel-levin** [dotnet/coreclr#23741]
Hi all,

I am trying to generate the CMake cache with Visual Studio 2017. My CMakeSettings.json file looks like:

```
{
  "configurations": [
    {
        "name": "x64-Debug",
        "generator": "Visual Studio 15 2017 Win64",
        "configurationType": "Debug",
        "inheritEnvironments": [
            "msvc_x64_x64"
        ],
        "buildRoot": "${env.USERPROFILE}\\CMakeBuilds\\${workspaceHash}\\build\\${name}",
        "installRoot": "${env.USERPROFILE}\\CMakeBuilds\\${workspaceHash}\\install\\${name}",
        "cmakeCommandArgs": "",
        "buildCommandArgs": "",
        "ctestCommandArgs": "",
        "variables": [
            {
                "name": "CLR_CMAKE_HOST_ARCH",
                "value": "x64"
            }
        ]
    }
  ]
}

```

but the following error is persistent:

```
CMake Error at src/vm/CMakeLists.txt:921 (add_subdirectory):
  add_subdirectory given source
  "C:/Users/Daniel/CMakeBuilds/b4325264-04d4-463f-aff8-b12da732d655/build/x64-Debug/Eventing/eventpipe"
  which is not an existing directory.		C:\Users\Daniel\source\repos\coreclr\src/vm/CMakeLists.txt	921
```

The `/build/$name/Eventing/eventpipe` directory is not being created. Please can someone help me find out where or how it should be created? If I modify the `src/vm/CMakeLists.txt:921` to create the directory (so the subdirectory failure does not occur) then it still does not work, reporting that `CMakeLists.txt` cannot be found in that directory.

**Comment by RussKeldorph** [comment]
FWIW, I don't think we put much effort into working with VS like this, so YMMV.  I think https://github.com/dotnet/coreclr/blob/29dc90c13a5d091c21e21e4ba5d3c4bcd3eab18d/build.cmd#L450 is related, but I'm not terribly familiar with it.  @brianrob may know more.

**Comment by brianrob** [comment]
Yup, there are a handful of directories/files that are created as part of the build.  The various CMake files reference them, but they don't get created until build.cmd is invoked.  This is one such place.

**Comment by jkoritzinsky** [comment]
Issue #2486 is tracking moving the EventPipe generated directories from being geenrated in a python script before build to being built in CMake.

**Comment by janvorli** [comment]
@daniel-levin what are you trying to achieve? Our build.cmd and src\pal\tools\gen-buildsys-win.bat scripts are doing quite a lot of things to prepare parameters for cmake invocation, so replicating it is a bit complex.
If you run build.cmd, it will automatically generate VS sln / project files for the whole coreclr, so you can open those in VS after that.

**Comment by daniel-levin** [comment]
@janvorli I am trying to build and run coreclr and associated programs from VS2017. I thought I would use it in the configuration alluded to by @RussKeldorph. Thank you for letting my know that the sln files were generated by the build process. I am still having issues running ildasm from VS (permissioning problems, I cannot execute the binaries I build from VS) but I will try solve these before commenting again.
